### ISSUE TITLE
Where a command returns multiple tables, and the last table has no rows, it is not included in the DataAdapter

### ISSUE THREAD

**Issue by badbod99** [mysql-net/MySqlConnector#524]
I'm using MySqlDataAdapter.Fill to fill a DataSet like this:

```C#
public static DataSet RetrieveDataSet(string connectionString, string storedProcedure, params MySqlParameter[] parameters)
        {
            using (MySqlConnection conn = new MySqlConnection(connectionString))
            {
                conn.Open();

                using (MySqlCommand comm = conn.CreateCommand())
                {
                    comm.CommandText = storedProcedure;
                    comm.CommandType = CommandType.StoredProcedure;
                    if (parameters != null && parameters.Length > 0)
                    {
                        comm.Parameters.AddRange(parameters);
                    }

                    using (MySqlDataAdapter adapter = new MySqlDataAdapter(comm))
                    {
                        DataSet data = new DataSet();
                        adapter.Fill(data);
                        return data;
                    }
                }
            }
        }
```

My stored procedure returns 4 result sets.  In the case that result set 4 (the last one) is empty, the data.Tables.Count equals 3.  In the case that result set 4 has rows, data.Tables.Count equals 4.  In the case that result set 2 has no rows, data.Tables.Count equals 4 still.

So it seems like if the last result set is blank, it is not being included as a table in the dataset.  My guess is that this is a for loop error, but I couldn't identify it from the code of MySqlDataAdapter.

**Comment by bgrainger** [comment]
Can repro with the test data set up in `StoredProcedureFixture`:

```csharp
using (var command = new MySqlCommand("multiple_result_sets", connection))
{
	command.CommandType = CommandType.StoredProcedure;
	command.Parameters.AddWithValue("pivot", 8);
	using (MySqlDataAdapter adapter = new MySqlDataAdapter(command))
	{
		DataSet data = new DataSet();
		adapter.Fill(data);
		Console.WriteLine(data.Tables.Count); // prints 1, should be 2
	}
}
```

**Comment by bgrainger** [comment]
Probably the underlying problem:

```csharp
using (var reader = command.ExecuteReader())
{
	while (reader.Read())
	{
	}
	
	Console.WriteLine(reader.NextResult()); // false, should be true
}
```

**Comment by bgrainger** [comment]
Which may be related to the fix for #135.

**Comment by bgrainger** [comment]
Fixed in [0.42.2](https://github.com/mysql-net/MySqlConnector/releases/tag/0.42.2).
